### PR TITLE
.gitignore: Drop auto-generated content

### PR DESCRIPTION
**Pull request by wking**
From [the docs][1]:

> Which file to place a pattern in depends on how the pattern is meant to be used.
>
> * Patterns which should be version-controlled and distributed to other repositories via clone (i.e., files that all developers will want to ignore) should go into a `.gitignore` file.
> * ...
> * Patterns which a user wants Git to ignore in all situations (e.g., backup or temporary files generated by the user's editor of choice) generally go into a file specified by `core.excludesFile` in the user's `~/.gitconfig`. Its default value is `$XDG_CONFIG_HOME/git/ignore`. If `$XDG_CONFIG_HOME` is either not set or empty, `$HOME/.config/git/ignore` is used instead.

So editor files are not appropriate for the project's versioned-controlled `.gitignore`.  We could keep the Go stuff, but it's not clear to me if we have tooling that generates any of those outside of the temporary build directories which already have manually-curated `.gitignore` entries.

[1]: https://git-scm.com/docs/gitignore#_description

### CODE REVIEW

**Review by PratikMahajan — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,77 +1,3 @@
...
-# Binaries for programs and plugins
-*.exe
-*.exe~
-*.dll
-*.so
-*.dylib
-# Test binary, build with 'go test -c'
-*.test
-# Output of the go coverage tool, specifically when used with LiteIDE
-*.out
```
[the remark this answers is not in the document]
Comment (wking) [reply .gitignore:63]: Is that something we run?

**Comment by wking** [comment]
Gitignore change unrelated to broken e2e.

/override ci/prow/operator-e2e

**Comment by wking** [comment]
[images][1]:

```
error: build error: failed to pull image: After retrying 2 times, Pull image still failed due to error: while pulling "docker://registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.7" as "registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.7": Error initializing source docker://registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.7: Error reading manifest rhel-8-golang-1.15-openshift-4.7 in registry.svc.ci.openshift.org/ocp/builder: unknown: unable to pull manifest from registry.ci.openshift.org/ocp/builder:latest: manifest unknown: manifest unknown
```

/test images

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cincinnati-operator/60/pull-ci-openshift-cincinnati-operator-master-images/1372257576611221504#1:build-log.txt%3A51

**Comment by wking** [comment]
/override ci/prow/images

But we'll need to dig into why these are repeatedly failing.

**Pull request merged**